gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java

```diff
@@ -1311,6 +1312,25 @@ public void putAssignmentsAndCategoryItemsInMatrix(Map<String, GbStudentGradeInf
 				// get the group membership for the students
 				final Map<String, List<String>> groupMembershipsMap = getGroupMemberships();
 
+				//Pair group <-> category
+				Map <Long, List<String>> permByCat = new HashMap<>();
+
+				for (final PermissionDefinition permission : permissions) {
+					final Long permissionCategoryId = permission.getCategoryId() !=null ? permission.getCategoryId() : -1L;
+					final String permissionGroupReference = permission.getGroupReference() !=null ? permission.getGroupReference() : "all";
+
+					//permissions over all categories and grades
+					if (Long.valueOf(-1L).equals(permissionCategoryId) && "all".equals(permissionGroupReference)) {
```

Comment:
Where does "all" come from? Could it be in a final String somewhere, a constant?